locust/test/test_web.py

```diff
@@ -1082,18 +1083,36 @@ def tearDown(self):
         self.web_ui.stop()
         self.runner.quit()
 
-    def test_index_with_basic_auth_enabled_correct_credentials(self):
-        self.assertEqual(
-            200, requests.get("http://127.0.0.1:%i/?ele=phino" % self.web_port, auth=("john", "doe")).status_code
-        )
+    def test_index_with_web_auth_enabled_valid_user(self):
+        class User(UserMixin):
+            def __init__(self):
+                self.username = "test_user"
 
-    def test_index_with_basic_auth_enabled_incorrect_credentials(self):
-        self.assertEqual(
-            401, requests.get("http://127.0.0.1:%i/?ele=phino" % self.web_port, auth=("john", "invalid")).status_code
-        )
+            def get_id(self):
+                return self.username
+
+        def load_user(id):
+            return User()
+
+        self.web_ui.login_manager.request_loader(load_user)
+
+        response = requests.get("http://127.0.0.1:%i" % self.web_port)
+        d = pq(response.content.decode("utf-8"))
+
+        self.assertNotIn("authArgs", str(d))
+        self.assertIn("templateArgs", str(d))
```

Comment:
No the first test basically simulates as if a user is already signed in, and that the app should then be returned. The 2nd test simulates a user that is not signed in and expects the log-in page to be returned.

We can't really call `login_user` from Flask login because we are not in a Flask context. I was thinking this might be the simplest way to provide some basic tests without testing Flask login or testing what should be implemented by the user